src/dune_engine/source_tree.ml

```diff
@@ -519,17 +541,14 @@ end = struct
 
   let get_vcs ~default:vcs ~readdir:{ Readdir.path; files; dirs } =
     match
-      match
-        List.find_map dirs ~f:(fun (name, _, _) -> Vcs.Kind.of_filename name)
-      with
-      | Some kind -> Some kind
-      | None -> Vcs.Kind.of_dir_contents files
+      Vcs.Kind.of_dir_contents
```

Comment:
I suppose we can also just pass the directories without the files here.

Reply:
Not really as you can have a `.git` file that contains the path to the git directory (git allows that)